### PR TITLE
misc, stdlib: Update documentation to adhere to RST formatting.

### PR DESCRIPTION
**Pull request by ivanaamit**
This PR updates files in `src/python` to adhere to reStructuredText formatting.

### CODE REVIEW

File: src/python/gem5/resources/resource.py [hunk 1]
```diff
@@ -824,32 +825,37 @@ def obtain_resource(
...
-    thrown. True by default.
+                               resource is to be stored. If this parameter is not
+                               set, it will set to the environment variable
+                               ``GEM5_RESOURCE_DIR``. If the environment is not set
+                               it will default to ``~/.cache/gem5`` if available,
+                               otherwise the CWD.
+    			                *Note*: This argument is ignored if the ``to_path``
+    			               parameter is specified.
+    :param download_md5_mismatch: If the resource is present, but does not have
+                                  the correct md5 value, the resoruce will be
```
Comment (Harshil2107) [inline src/python/gem5/resources/resource.py]:
```suggestion
                                  the correct md5 value, the resource will be
```

This is not related to the reformatting, but resource is misspelled as resoruce. I think we can change change this in this PR as it updates the documentation.

File: src/python/gem5/resources/resource.py [hunk 2]
```diff
@@ -1108,9 +1117,9 @@ def Resource(
     This function was created to maintain backwards compability for v21.1.0
```
Comment (Harshil2107) [inline src/python/gem5/resources/resource.py]:
```suggestion
    This function was created to maintain backwards compatibility for v21.1.0
```
This is not related to the reformat but compatibility is misspelled as compability.

File: src/python/gem5/resources/resource.py [hunk 3]
```diff
@@ -1108,9 +1117,9 @@ def Resource(
...
     and prior releases of gem5 where `Resource` was a class.
 
-    In the interests of gem5-resource specialization, the `Resource` class
-    has been dropped. Instead users are advized to use the `obtain_resource`
-    function which will return the correct `AbstractResource` implementation.
+    In the interests of gem5-resource specialization, the ``Resource`` class
+    has been dropped. Instead users are advized to use the ``obtain_resource``
```
Comment (Harshil2107) [inline src/python/gem5/resources/resource.py]:
```suggestion
    has been dropped. Instead users are advised to use the ``obtain_resource``
```
This is not related to the reformat but advised is misspelled as advized.

**Review by Harshil2107 — CHANGES_REQUESTED** [review]
The documentation changes look mostly good to me. There are some spelling errors that we might as well fix as we are updating documentation.

**Comment by ivanaamit** [comment]
Good catch, @Harshil2107! I fixed the errors.

**Review by Harshil2107 — APPROVED** [review]
lgtm

**Pull request merged**